packages/fxa-auth-server/test/local/subhub/updates.js

```diff
@@ -0,0 +1,119 @@
+
+const EventEmitter = require('events').EventEmitter;
+const { mockDB, mockLog } = require('../../mocks');
+const subhubUpdates = require('../../../lib/subhub/updates');
+
+const mockDeliveryQueue = new EventEmitter();
+mockDeliveryQueue.start = function start() { };
+
+const baseMessage = {
+  uid: '1e2122ba',
+  subscriptionId: '1234',
+  productName: 'fx_pro',
+  eventId: 'st_ev_1234',
+  eventCreatedAt: 1557265730749,
+  messageCreatedAt: 1557265730949
+};
+
+
+function mockMessage(messageOverrides) {
+  const message = Object.assign({}, baseMessage, messageOverrides);
+  message.del = sinon.spy();
+  return message;
+}
+
+function mockSubHubUpdates(log, db) {
+  return subhubUpdates(log)(mockDeliveryQueue, db);
+}
+
+describe('subhub updates', () => {
+  let db;
+  let log;
+
+  beforeEach(() => {
+    db = mockDB();
+    log = mockLog();
+  });
+
+  it(
+    'should log validation errors',
+    async () => {
+      await mockSubHubUpdates(log, db).handleSubHubUpdates(mockMessage({subscriptionId: null, active: true}));
+      assert.equal(log.error.callCount, 1);
+    }
+  );
+
+  it(
+    'should activate an account',
+    async () => {
+      await mockSubHubUpdates(log, db).handleSubHubUpdates(mockMessage({active: true}));
+      // FIXME: figure out what side effect we expect
+      assert.equal(log.error.callCount, 0, `Got error: ${log.error.lastCall ? JSON.stringify(log.error.lastCall.args) : ''}`);
+      assert.calledWithExactly(
+        db.createAccountSubscription,
+        baseMessage.uid, baseMessage.subscriptionId, baseMessage.productName, baseMessage.eventCreatedAt);
+    }
+  );
+
+  it(
+    'should de-activate an account',
+    async () => {
+      await mockSubHubUpdates(log, db).handleSubHubUpdates(mockMessage({active: false}));
+      // FIXME: figure out what side effect we expect
+      assert.calledWithExactly(db.deleteAccountSubscription, baseMessage.uid, baseMessage.subscriptionId);
+      assert.equal(log.error.callCount, 0);
+    }
+  );
+
+  it(
+    'should ignore an UNSUBSCRIBE message that is out of order',
+    async () => {
+      const message = mockMessage({eventCreatedAt: baseMessage.eventCreatedAt, active: false});
+      db.getAccountSubscription = sinon.spy(
+        async (uid, subscriptionId) => {
+          return {
+            uid,
+            subscriptionId,
+            productName: message.productName,
+            // It's a subscription FROM THE FUTURE!
+            createdAt: message.eventCreatedAt + 1000,
+          };
+        }
+      );
+      await mockSubHubUpdates(log, db).handleSubHubUpdates(message);
+      assert.equal(db.getAccountSubscription.callCount, 1, 'db.getAccountSubscription() should be called');
+      assert.equal(db.deleteAccountSubscription.callCount, 0, 'db.deleteAccountSubscription() should not be called');
+    }
+  );
+
+  // FIXME: because we don't keep timestamps of unsubscribes, we can't implement this:
+  it.skip(
```

Comment:
If we land it like this, I'll create a followup ticket to fix this. (I'm guessing it won't be high priority, but should be acknowledged.)

Reply:
See #1129